pkg/apis/openebs.io/v1alpha1/common_volume_properties.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2018 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+// CapacitySpec wraps the Capacity property
+//
+// NOTE: It is intended to be used as embedded struct
+type CapacitySpec struct {
+	Capacity string `json:"capacity"`
+}
+
+// TargetIPSpec wraps the TargetIP property
+//
+// NOTE: It is intended to be used as embedded struct
+type TargetIPSpec struct {
+	TargetIP string `json:"targetIP"`
+}
```

Comment:
1. I think this is not needed ...we are making it complex and extra nesting of structs in all ISCSI object. Having a single struct `ISCSISPEC` which will contains all the required volumesource fields will be sufficient.

Reply:
This has to be done because `CStorVolumeReplicaSpec` contains `TargetIP` as well as `Capacity`. Not doing so will again result in scattered properties.